twoliter/src/cmd/build.rs

```diff
@@ -25,10 +33,94 @@ pub(crate) struct BuildVariant {
     /// The architecture to build for.
     #[clap(long = "arch", default_value = "x86_64")]
     arch: String,
+
+    /// The variant to build.
+    variant: String,
 }
 
 impl BuildVariant {
     pub(super) async fn run(&self) -> Result<()> {
-        Ok(())
+        let project = project::load_or_find_project(self.project_path.clone()).await?;
+        let token = project.token();
+        let tempdir = tools_tempdir()?;
+        install_tools(&tempdir).await?;
+        let makefile_path = tempdir.path().join("Makefile.toml");
+        let packages_dir =
+            TempDir::new().context("Unable to create a tempdir for Twoliter's packages")?;
+
+        let sdk_container = DockerContainer::new(
+            format!("sdk-{}", token),
+            project
+                .sdk(&self.arch)
+                .context("The project was missing an sdk")?
```

Comment:
nit: maybe add more context:
```suggestion
                .context(format!("No SDK defined in {} for {}", &self.project_path.display(), &self.arch))?
```